apps/election-manager/src/components/ExportElectionBallotPackageModalButton.tsx

```diff
@@ -0,0 +1,301 @@
+   */
+  const onRendered = useCallback(async () => {
+    if (state.type !== 'RenderBallot') {
+      throw new Error(
+        `unexpected state '${state.type}' found during onRendered callback`
+      )
+    }
+
+    const {
+      ballotStyleId,
+      precinctId,
+      locales,
+      isLiveMode,
+    } = state.currentBallotConfig
+    const path = getBallotPath({
+      ballotStyleId,
+      election,
+      electionHash,
+      precinctId,
+      locales,
+      isLiveMode,
+    })
+    const data = await window.kiosk!.printToPDF()
+    await state.archive.file(path, Buffer.from(data))
+    setState(workflow.next)
+  }, [election, electionHash, state])
+
+  const closeModal = () => {
+    setIsModalOpen(false)
+    setState(workflow.init(election, electionHash, electionLocaleCodes))
+  }
+
+  const now = new Date()
+  const defaultFileName = generateFilenameForBallotExportPackage(
+    electionDefinition!,
+    now
+  )
+
+  // Callback to open the file dialog.
+  const openFileDialog = async () => {
+    if (state.type !== 'ArchiveBegin') {
+      throw new Error(
+        `unexpected state '${state.type}' found during openFileDialog callback`
+      )
+    }
+    try {
+      await state.archive.begin({
+        defaultPath: defaultFileName,
+        filters: [{ name: 'Archive Files', extensions: ['zip'] }],
+      })
+      await state.archive.file('election.json', electionData)
+      await state.archive.file(
+        'manifest.json',
+        JSON.stringify({ ballots: state.ballotConfigs }, undefined, 2)
+      )
+
+      setState(workflow.next)
+    } catch (error) {
+      setState(workflow.error(state, error))
+    }
+  }
+
+  let mainContent = null
+  let primaryButton = null
+  let disableCancel = false
+
+  switch (state.type) {
+    case 'Init': {
+      mainContent = <Loading />
+      disableCancel = true
+      break
+    }
+
+    case 'ArchiveBegin':
+      switch (usbDriveStatus) {
+        case UsbDriveStatus.absent:
+        case UsbDriveStatus.notavailable:
+        case UsbDriveStatus.recentlyEjected:
+          mainContent = (
+            <Prose>
+              <h1>No USB Drive Detected</h1>
+              <p>
+                <USBImage src="usb-drive.svg" alt="Insert USB Image" />
+                Please insert a USB stick in order to export the ballot
+                configuration.
+              </p>
+            </Prose>
+          )
+          break
+        case UsbDriveStatus.ejecting:
+        case UsbDriveStatus.present:
+          mainContent = <Loading />
+          disableCancel = true
+          break
+        case UsbDriveStatus.mounted: {
+          // TODO(caro): Update this to just write the file to the USB stick once the APIS in kiosk-browser exist.
+          primaryButton = (
+            <Button onPress={openFileDialog} primary>
+              Export
+            </Button>
+          )
+          mainContent = (
+            <Prose>
+              <h1>USB Drive Detected!</h1>
```

Comment:
This should be the title of the current state. While it may be good to provide status that the USB Drive has been detected, that messaging should appear and then perhaps disappear. In the flow where a USB Drive is already mounted, there is no reason to note that the USB Drive has been detected. The fact that the screen is ready to go should be enough indication that the user can proceed with the export.
```suggestion
              <h1>Export Ballot Package</h1>
```

Reply:
Addressed in https://github.com/votingworks/vxsuite/pull/97/commits/98fd424c5290ff406ebe5302330c8bdbb381072a